src/app/backend/resource/pod/detail.go

```diff
@@ -183,18 +160,21 @@ func getPodCreator(client k8sClient.Interface, creatorAnnotation string, nsQuery
 		return nil, err
 	}
 
-	events := <-channels.EventList.List
-	if err := <-channels.EventList.Error; err != nil {
-		return nil, err
-	}
 	reference := serializedReference.Reference
 
-	controller, err := toPodController(client, reference, pods.Items, events.Items, heapsterClient)
+	rc, _ := owner.NewResourceController(reference, client)
+
+	controller := rc.Get(pods.Items)
+
+	fmt.Println(controller)
+
+	// TODO
```

Comment:
I guess this TODO means "improve error handling", but I'm not sure (particularly as there's a blank line after it).
Can you specify its meaning?